app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListInteractor.kt

```diff
@@ -166,10 +190,35 @@ class ConnectionsListInteractor(
 
     private fun notifyDatasetChanges() {
         contract?.onDatasetChanged(
-            connections = richConnections.map { it.connection }.sortedBy { it.createdAt },
-            consents = allConsents
+            connections = richConnections.map {
+                it.connection
+            }.sortedBy { it.createdAt },
+            consents = allConsents,
+            providerLogoUrl = providerLogoUrl
         )
     }
+
+    override fun onShowConnectionConfigurationSuccess(
+        result: ConfigurationDataV2
+    ) {
+        richConnections.map {
+            if (result.providerLogoUrl == it.connection.logoUrl) return
+            else {
+                this.providerLogoUrl = listOf(result.providerLogoUrl)
+                connectionsRepository.getAllConnections().map { connection ->
+                    this.providerLogoUrl.map {
+                        connection.logoUrl = it
+                        connectionsRepository.saveModel(connection)
+                        notifyDatasetChanges()
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onShowConnectionConfigurationFailed(error: ApiErrorData) {
+        contract?.onError(error)
```

Comment:
do we need to show configuration fetch errors?

Reply:
I think yes, standard good behavior